packages/react-components/react-shared-contexts/src/ProviderContext/ProviderContext.ts

```diff
@@ -11,17 +11,20 @@ export type ProviderContextValue = {
 /**
  * @internal
  */
-// eslint-disable-next-line @fluentui/no-context-default-value
-const ProviderContext = React.createContext<ProviderContextValue>({
+const ProviderContext = React.createContext<ProviderContextValue | undefined>(
+  undefined,
+) as React.Context<ProviderContextValue>;
+
+const providerContextDefaultValue: ProviderContextValue = {
   targetDocument: typeof document === 'object' ? document : undefined,
-  dir: 'ltr',
-});
+  dir: 'ltr' as 'ltr',
```

Comment:
Is it need at all?

P.S. May be `as const`?

```suggestion
  dir: 'ltr' as const,
```